### PR TITLE
Add quadratic objective support

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

**Comment by amontoison** [comment]
The last feature that is missing is quadratic constraints. But some updates in `NLPModels.jl` are required before we can add them here.

File: src/moi_nlp_model.jl [hunk 1]
```diff
@@ -25,17 +25,20 @@ function MathOptNLPModel(jmodel :: JuMP.Model; name :: String="Generic")
...
+  else
+    obj = parser_objective_MOI(moimodel, nvar)
+  end
 
   ncon = nlin + nnln
   lcon = vcat(lin_lcon, nl_lcon)
   ucon = vcat(lin_ucon, nl_ucon)
   nnzj = lin_nnzj + nl_nnzj
-  nnzh = nl_nnzh
+  nnzh = length(obj.vals) + nl_nnzh
```
[the remark this answers is not in the document]
Comment (amontoison) [reply src/moi_nlp_model.jl]: `length(eval.objective.hess_I)` is the number of `nnz` in the objective Hessian if it's nonlinear. `length(obj.vals)` is the number of `nnz` in the objective Hessian if it's quadratic.

if the objective is quadratic, `eval.has_nlobj = false`.

[the remark this answers is not in the document]
Comment (dpo) [reply src/moi_nlp_model.jl]: Ah so quadratic is not nonlinear ;-)

[the remark this answers is not in the document]
Comment (amontoison) [reply src/moi_nlp_model.jl]: Linear / Quadratic -> `@objective`, so `eval.has_nlobj = false`. But it should be `eval.has_obj` in practice. One day they will update the NLP part of JuMP...

**Comment by dpo** [comment]
Thank you!

**Pull request merged**